### PR TITLE
Add multi threading to MCP steps that can be run in parallel

### PR DESCRIPTION
**Pull request by TaranFurnell**
- Adds threading to allow CPAS and the IAP reference model to be done in parallel (CPAS is SAS UUT side, IAP is test harness side)
- Adds threading to allow FAD generation and IAP aggregate interference check to be done in parallel (FAD generation is SAS UUT side, aggregate interference check is test harness side)
- Adds threading to allow DPA movelist calculation and IAP aggregate interference check to be done in parallel (Both are test harness side but do not impact one another, and are potentially long running)

### CODE REVIEW

File: src/harness/full_activity_dump_helper.py [hunk 1]
```diff
@@ -80,22 +79,18 @@ def _triggerFullActivityDumpAndWaitUntilComplete(sas, sas_admin, ssl_cert,
   """
   request_time = datetime.utcnow().replace(microsecond=0)
   sas_admin.TriggerFullActivityDump()
-  signal.signal(signal.SIGALRM,
-                lambda signum,
-                frame: (_ for _ in ()).
-                  throw(EnvironmentError('Full Activity Dump '
-                                         'generation in SAS UUT timed out')))
-  # Timeout after 2 hours if it's not completed.
-  signal.alarm(7200)
```
Comment (TaranFurnell) [inline src/harness/full_activity_dump_helper.py]: Change required to execute the code in a thread that is not the main thread.

File: src/harness/sas_testcase.py [hunk 2]
```diff
@@ -228,13 +227,8 @@ def TriggerDailyActivitiesImmediatelyAndWaitUntilComplete(self):
     - If the status is not changed within 2 hours it will throw an exception.
     """
     self._sas_admin.TriggerDailyActivitiesImmediately()
-    signal.signal(signal.SIGALRM,
-                  lambda signum, frame:
-                  (_ for _ in ()).throw(
-                      Exception('Daily Activity Check Timeout')))
-
```
Comment (TaranFurnell) [inline src/harness/sas_testcase.py]: As above.

File: src/harness/testcases/WINNF_FT_S_MCP_testcase.py [hunk 3]
```diff
@@ -353,8 +371,9 @@ def executeSingleMCPIteration(self, iteration_content):
                                           test_harness.getSasTestHarnessInterface()))
 
     # Step 8: Trigger CPAS and wait until completion.
-    logging.info('Step 8: Triggering CPAS.')
-    self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
+    logging.info("Step 8: Triggering CPAS.")
+    cpas = ParallelCpas(self)
+    cpas.start()
```
Comment (TaranFurnell) [inline src/harness/testcases/WINNF_FT_S_MCP_testcase.py]: CPAS is still triggered at the same time as previously, this simply allows other work to be completed while it executes.

File: src/harness/testcases/WINNF_FT_S_MCP_testcase.py [hunk 4]
```diff
@@ -403,8 +425,9 @@ def executeSingleMCPIteration(self, iteration_content):
     # For the set of protected entity the interference value is calculated
     # using both IAP and aggregate interference reference models, checked against
     # the comparison value
-    logging.info('Step 12: performing aggregate interference check.')
-    self.performAggregateInterferenceCheck()
+    logging.info("Step 12: performing aggregate interference check.")
+    aggregate_interference_check = ParallelAggregateInterferenceCheck(self)
+    aggregate_interference_check.start()
```
Comment (TaranFurnell) [inline src/harness/testcases/WINNF_FT_S_MCP_testcase.py]: The same for aggregate interference check.

**Comment by jni2000** [comment]
we test run these changes, overall looks good and run without error and able to execute test cases. However, we do not have sufficient data to thoroughly validate the results. Can CommScope help do additional validations or give opinions ?

**Pull request merged**